### ISSUE TITLE
CachingServiceInstanceListSupplier bug when using redisCachemanager

### ISSUE THREAD

**Issue by fanticat** [spring-cloud/spring-cloud-commons#625]
When we have data-redis on classpath, a redisCachemanager will be auto configed, then CachingServiceInstanceListSupplier use this cache manage to cache the EurekaServiceInstance, but EurekaServiceInstance can not be serialized by Jdkserializer, cause EurekaServiceInstance has not implement Serialize, if we use jackson serializer, it successful, but can not be deserialized, because no default construction in EurekaServiceInstance

**Comment by spencergibb** [comment]
If the issue is to not use the redis cache manager the issue belongs here, if it is to fix `EurekaServiceInstance` for serialization, it belongs in spring-cloud-netflix.

My first thought is that it probably shouldn't use persistent caching, just in memory.

**Comment by spencergibb** [comment]
ping @OlgaMaciaszek

**Comment by fanticat** [comment]
It's a distributed system, and configed a redisCachemanager as primary bean to cache business data, as your suggestion,  I need to config another  in memory cache manager, and inject myown CachingServiceInstanceListSupplier

**Comment by cashZhang27** [comment]
I've had this problem, too.

**Comment by cashZhang27** [comment]
@spencergibb I also encountered this problem when I was using redis reactive and SC LoadBalancer, the first time the A service called the B service using WebClient, but the second call found that the data in redis could not be deserialized.

springcloud version Hoxton.RC2
springboot version 2.2.1.RELEASE

```json
[
  "java.util.ArrayList",
  [
    [
      "org.springframework.cloud.netflix.eureka.EurekaServiceInstance",
      {
        "instance": [
          "com.netflix.appinfo.InstanceInfo",
          {
            "instanceId": "localhost:cloud-webflux-say:8091",
            "app": "CLOUD-WEBFLUX-SAY",
            "appGroupName": null,
            "ipAddr": "192.168.46.1",
            "sid": "na",
            "port": 8091,
            "securePort": 443,
            "homePageUrl": "http://localhost:8091/",
            "statusPageUrl": "http://localhost:8091/info",
            "healthCheckUrl": "http://localhost:8091/health",
            "secureHealthCheckUrl": null,
            "vipAddress": "cloud-webflux-say",
            "secureVipAddress": "cloud-webflux-say",
            "countryId": 1,
            "dataCenterInfo": {
              "@class": "com.netflix.appinfo.InstanceInfo$DefaultDataCenterInfo",
              "name": "MyOwn"
            },
            "hostName": "localhost",
            "status": "UP",
            "overriddenStatus": "UNKNOWN",
            "leaseInfo": [
              "com.netflix.appinfo.LeaseInfo",
              {
                "renewalIntervalInSecs": 5,
                "durationInSecs": 90,
                "registrationTimestamp": 1574173514534,
                "lastRenewalTimestamp": 1574174605936,
                "evictionTimestamp": 0,
                "serviceUpTimestamp": 1574173514534
              }
            ],
            "isCoordinatingDiscoveryServer": false,
            "metadata": [
              "java.util.Collections$SynchronizedMap",
              {
                "management.port": "8091"
              }
            ],
            "lastUpdatedTimestamp": 1574173514534,
            "lastDirtyTimestamp": 1574173514479,
            "actionType": "ADDED",
            "asgName": null,
            "statusPageRelativeUrl": null,
            "statusPageExplicitUrl": null,
            "healthCheckRelativeUrl": null,
            "healthCheckSecureExplicitUrl": null,
            "vipAddressUnresolved": null,
            "secureVipAddressUnresolved": null,
            "healthCheckExplicitUrl": null,
            "isSecurePortEnabled": false,
            "isUnsecurePortEnabled": true,
            "isInstanceInfoDirty": false
          }
        ],
        "host": "localhost",
        "port": 8091,
        "metadata": [
          "java.util.Collections$SynchronizedMap",
          {
            "management.port": "8091"
          }
        ],
        "instanceId": "localhost:cloud-webflux-say:8091",
        "serviceId": "CLOUD-WEBFLUX-SAY",
        "secure": false,
        "uri": "http://localhost:8091",
        "instanceInfo": [
          "com.netflix.appinfo.InstanceInfo",
          {
            "instanceId": "localhost:cloud-webflux-say:8091",
            "app": "CLOUD-WEBFLUX-SAY",
            "appGroupName": null,
            "ipAddr": "192.168.46.1",
            "sid": "na",
            "port": 8091,
            "securePort": 443,
            "homePageUrl": "http://localhost:8091/",
            "statusPageUrl": "http://localhost:8091/info",
            "healthCheckUrl": "http://localhost:8091/health",
            "secureHealthCheckUrl": null,
            "vipAddress": "cloud-webflux-say",
            "secureVipAddress": "cloud-webflux-say",
            "countryId": 1,
            "dataCenterInfo": {
              "@class": "com.netflix.appinfo.InstanceInfo$DefaultDataCenterInfo",
              "name": "MyOwn"
            },
            "hostName": "localhost",
            "status": "UP",
            "overriddenStatus": "UNKNOWN",
            "leaseInfo": [
              "com.netflix.appinfo.LeaseInfo",
              {
                "renewalIntervalInSecs": 5,
                "durationInSecs": 90,
                "registrationTimestamp": 1574173514534,
                "lastRenewalTimestamp": 1574174605936,
                "evictionTimestamp": 0,
                "serviceUpTimestamp": 1574173514534
              }
            ],
            "isCoordinatingDiscoveryServer": false,
            "metadata": [
              "java.util.Collections$SynchronizedMap",
              {
                "management.port": "8091"
              }
            ],
            "lastUpdatedTimestamp": 1574173514534,
            "lastDirtyTimestamp": 1574173514479,
            "actionType": "ADDED",
            "asgName": null,
            "statusPageRelativeUrl": null,
            "statusPageExplicitUrl": null,
            "healthCheckRelativeUrl": null,
            "healthCheckSecureExplicitUrl": null,
            "vipAddressUnresolved": null,
            "secureVipAddressUnresolved": null,
            "healthCheckExplicitUrl": null,
            "isSecurePortEnabled": false,
            "isUnsecurePortEnabled": true,
            "isInstanceInfoDirty": false
          }
        ],
        "scheme": null
      }
    ]
  ]
]
```
```java
@Configuration
@EnableConfigurationProperties(RedisProperties.class)
public class RedisConfiguration extends CachingConfigurerSupport {

  @Bean
  public LettuceConnectionFactory redisConnectionFactory(RedisProperties redisProperties) {
    RedisStandaloneConfiguration redisStandaloneConfiguration = new RedisStandaloneConfiguration();
    redisStandaloneConfiguration.setHostName(redisProperties.getHost());
    redisStandaloneConfiguration.setDatabase(redisProperties.getDatabase());
    redisStandaloneConfiguration.setPort(redisProperties.getPort());
    redisStandaloneConfiguration.setPassword(redisProperties.getPassword());
    return new LettuceConnectionFactory(redisStandaloneConfiguration);
  }

  @Value("${spring.application.name}")
  private String applicationName;

  @Bean
  public RedisCacheManager redisCacheManager(RedisConnectionFactory redisConnectionFactory) {

    Jackson2JsonRedisSerializer<Object> serializer = getObjectJackson2JsonRedisSerializer();
    RedisCacheWriter redisCacheWriter =
        RedisCacheWriter.nonLockingRedisCacheWriter(redisConnectionFactory);
    RedisCacheConfiguration redisCacheConfiguration =
        RedisCacheConfiguration.defaultCacheConfig()
            .serializeValuesWith(
                RedisSerializationContext.SerializationPair.fromSerializer(serializer));
    redisCacheConfiguration.prefixKeysWith(applicationName);

    return new RedisCacheManager(redisCacheWriter, redisCacheConfiguration);
  }

  @Bean
  ReactiveStringRedisTemplate reactiveRedisTemplate(ReactiveRedisConnectionFactory factory) {
    return new ReactiveStringRedisTemplate(factory);
  }

  /* =================================== private method start =================================== */

  private Jackson2JsonRedisSerializer<Object> getObjectJackson2JsonRedisSerializer() {
    Jackson2JsonRedisSerializer<Object> serializer =
        new Jackson2JsonRedisSerializer<>(Object.class);
    ObjectMapper objectMapper = new ObjectMapper();
    objectMapper.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.ANY);
    objectMapper.activateDefaultTyping(
        objectMapper.getPolymorphicTypeValidator(), ObjectMapper.DefaultTyping.NON_FINAL);
    serializer.setObjectMapper(objectMapper);
    return serializer;
  }

  /* =================================== private method end =================================== */

}

```
```xml

  <dependencies>
    <!-- Spring Cloud Starter Netflix Eureka Client -->
    <dependency>
      <groupId>org.springframework.cloud</groupId>
      <artifactId>spring-cloud-starter-netflix-eureka-client</artifactId>
      <exclusions>
        <exclusion>
          <artifactId>spring-cloud-starter-netflix-ribbon</artifactId>
          <groupId>org.springframework.cloud</groupId>
        </exclusion>
        <exclusion>
          <artifactId>ribbon-eureka</artifactId>
          <groupId>com.netflix.ribbon</groupId>
        </exclusion>
        <exclusion>
          <artifactId>spring-cloud-netflix-ribbon</artifactId>
          <groupId>org.springframework.cloud</groupId>
        </exclusion>
      </exclusions>
    </dependency>
    <!-- Webflux -->
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-webflux</artifactId>
    </dependency>
    <!-- CircuitBreaker -->
    <dependency>
      <groupId>org.springframework.cloud</groupId>
      <artifactId>
        spring-cloud-starter-circuitbreaker-reactor-resilience4j
      </artifactId>
    </dependency>
    <!-- Spring Boot Data Redis Starter -->
    <dependency>
      <groupId>org.springframework.boot</groupId>
      <artifactId>spring-boot-starter-data-redis-reactive</artifactId>
    </dependency>
    <dependency>
      <groupId>org.apache.commons</groupId>
      <artifactId>commons-pool2</artifactId>
    </dependency>
  </dependencies>

```

**Comment by xue8** [comment]
I have the same problem, did you solve it？

**Comment by cashZhang27** [comment]
> I have the same problem, did you solve it？

using Hoxton GA

**Comment by xue8** [comment]
> > I have the same problem, did you solve it？
> 
> using Hoxton GA

ok thank you

**Comment by spencergibb** [comment]
Fixed by #643
